### DISCUSSION TITLE
[About postprocessor][Accumulation values]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#17945] [Q&A Modules: General] [answered]
Dear users,

I am trying to find a way to implement accumulation values after each time step in my transient model.

The accumulated values should be used in the next time step to calculate residuals.

Can I use a postprocessor to do this?

Because I have tried to use auxkernels and material kernels before,
but I could not find a way to accumulate value in auxkernel.

I could accumulate values by material kernels but, it repeated inside of the residual calculation and showed me wrong values. (via _old_Name[_qp]).

**Comment by loganharbour** [comment, marked as answer]
Take a look at CumulativeValuePostprocessor.

Here's an example input: https://github.com/idaholab/moose/blob/next/test/tests/postprocessors/cumulative_value_postprocessor/cumulative_value_postprocessor.i, which is accumulating the number of nonlinear iterations from another postprocessor over time.

**Reply by coskrrb2002** [reply]
Dear loganharbour,

Thank you for your comment!
